src/bun.js/api/server.zig

```diff
@@ -1324,6 +1324,13 @@ pub const AnyRequestContext = struct {
         return .{ .tagged_pointer = Pointer.init(request_ctx) };
     }
 
+    pub fn getRemoteSocketId(self: AnyRequestContext) i32 {
+        if (self.tagged_pointer.as(HTTPServer.RequestContext).getNativeHandle()) |handle| {
```

Comment:
This would be needed for each type matching the tagged pointer, not just the one type here.